+adi/+AD7124/Base.m

```diff
@@ -66,7 +66,11 @@ function delete(obj)
             % Set device sampling rate
             obj.SampleRate = value;
             if obj.ConnectedToDevice
-                obj.setDeviceAttributeRAW('sampling_frequency', value);
+                for i=1:length(obj.channel_names)
```

Comment:
@commodo, Can we update the driver to make sure the no-OS implementation doesn't break with these changes?
In the tinyIIOd-based Firmware implementation, this attribute is treated as a device-level attribute and I see that these changes align with the new Linux drivers.

We can have a new property for channel-sampling-rate attribute naming it as SampleRateCh (or something of that sort or a boolean property that tells if it's a device level or channel level) and this logic can go there (toward its setter function). This way we can have this driver compatible for both the implementations and the attributes can be set accordingly in the example script.

Reply:
AFAICT, the no-Os drivers should be differential;
no idea how they show up in IIO

@buha @amiclaus    any ideas?

i'll need to see if i can make some time to check this